src/components/Board.vue

```diff
@@ -102,27 +107,19 @@
         methods: {
             start() {
                 this.playSound('gameStart');
-
                 if (this.gameRunning !== null) {
                     return;
                 }
 
-                this.isGameOver = false;
-
-                this.score = 0;
-
-                this.speed = 500;
-
-                // Resets grid, snake position, and snake movement variables back to their defaults
-                this.squares = this.resetGrid();
-                this.direction = DIRECTION.RIGHT;
-                this.snakeHead = SNAKE.HEAD_START;
-                this.snakeBody = SNAKE.BODY_START;
-                this.canChangeDirection = true;
-                this.resetSnake();
-
-                this.squares[this.getFoodRandomCoords()] = SNAKE.FOOD;
-
+                this.resetGame();
+                this.run();
+            },
+            reset(){
+                this.playSound('gameStart');
+                if (this.gameRunning == null || this.gameRunning === false) {
```

Comment:
This can be turned into if (!this.gameRunning) as it will check for undefined, null, or false